src/main/java/org/broadinstitute/hellbender/tools/htsgetreader/HtsgetResponse.java

```diff
@@ -53,26 +45,18 @@ public HtsgetClass getDataClass() {
         }
 
         /**
-         * Returns InputStream containing data of a single block from the htsget response
-         *
-         * Large blocks (those behind an http(s) URI) are first saved
-         * to a temp file that is deleted upon program exit
+         * Gets data from this block either from its base64 encoded data url or by making an http request
+         * @return InputStream of data from this block
          */
         public InputStream getData() {
             switch (this.getUri().getScheme()) {
                 case "http":
                 case "https":
                     final HttpGet get = new HttpGet(this.getUri());
                     this.getHeaders().forEach(get::addHeader);
-                    try (final CloseableHttpResponse resp = HttpUtils.getClient().execute(get)) {
-                        final Path outputFile = IOUtils.createTempPath("htsget-temp", "");
-                        try (final OutputStream ostream = Files.newOutputStream(outputFile);
-                             final InputStream istream = resp.getEntity().getContent()) {
-                            org.apache.commons.io.IOUtils.copy(istream, ostream);
-                        } catch (final IOException e) {
-                            throw new UserException("Could not write to temp file", e);
-                        }
-                        return Files.newInputStream(outputFile);
+                    try {
+                        final HttpResponse resp = HttpUtils.getClient().execute(get);
```

Comment:
Do you think we'll see timeouts due to these no longer aggressively downloading the blocks?

Reply:
It should make it less likely since they're only being requested once they need to be read, but if the read goes slowly and it's a large chunk it's still possible it could time out